app/addons/replication/api.js

```diff
@@ -41,6 +42,38 @@ export const getAuthHeaders = (username, password) => {
   };
 };
 
+export const getCredentialsFromUrl = (url) => {
+  const index = url.lastIndexOf('@');
+  if (index === -1) {
+    return {
+      username: '',
+      password: ''
+    };
+  }
+
+  const startIndex = url.startsWith("https") ? 8 : 7;
+  const rawCreds = url.slice(startIndex, index);
+  const colonIndex = rawCreds.indexOf(':');
+  const username = rawCreds.slice(0, colonIndex);
+  const password = rawCreds.slice(colonIndex + 1, rawCreds.length);
```

Comment:
Minor note but I think you could combine all of these variable instantiations and assignments in to a single const statement.